spark-bigquery-pushdown/pushdown_common_src/main/scala/com/google/cloud/spark/bigquery/pushdowns/BigQueryStrategy.scala

```diff
@@ -178,6 +179,27 @@ class BigQueryStrategy(expressionConverter: SparkExpressionConverter, expression
           }
         }
 
+      case UnionOperationExtractor(logicalPlanSeq) =>
+
+        /**
+         * Need to convert the list of logical plan to BigQuerySQLQuery
+         */
+        val children: Seq[BigQuerySQLQuery] = {
+          logicalPlanSeq.map { child =>
+            generateQueryFromPlan(child).get
+          }
+        }
+        val outputAttributes: Option[Seq[Attribute]] = Some(children.head.output)
```

Comment:
We don't need to send in `outputAttributes` and `visibleAttributes` from here itself. Let's only pass the required params `expressionConverter`, `expressionFactory`, `children` and `alias`

Reply:
We can compute outputAttributes and visibleAttributes in the UnionQuery class

Reply:
Addressed